### ISSUE TITLE
As an admin, I want library members to have unique identifiers that can be included for use in URLs in the public site.

### ISSUE THREAD

**Issue by rgmunson** [Princeton-CDH/mep-django#219]
## testing notes
- when you browse the list of members and load an individual page, the url in your browser location should have a "slug" (lower case letters with optional dashes) instead of a number; check at least one member detail page and member activities
  - check at least one pair of members with the same last name
  - check at least one set of names where the same name is repeated multiple times without a first name to differentiate (they should have numbers to distinguish them in the url)
- in django admin, person slug should be visible on the person edit page with help text to warn about not changing
- when you create a new person record, a preliminary slug should automatically be generated from the name you enter
- if you try to save a new person record with a slug that already exists, you should get an error

## dev notes
- [x] add slug to person
- [x] write migration to generate unique slugs
  - [x] use last name only if sufficiently unique
  - [x] use lastname-firstname if unique
  - [x] use lastname-1 etc if required to make unique
- [x] update view to use slug for url
- [x] update person absolute url method to use slug
- [x] make slugs and editable in admin
- [x] set slugs properly on new records

**Comment by rgmunson** [comment]
We're waiting to do this until the person merge work is complete.

**Comment by rlskoeser** [comment]
Split logic for old slugs and redirects out into #434

**Comment by clmahoney** [comment]
- YES, I see the slugs! 
- First I checked Sam Acheson, and the slug is: https://test-shakespeareandco.cdh.princeton.edu/members/acheson/ 
- Then I checked the Edels, and YES I see Bertha is edel-bertha-cohen (since we haven't adjusted her name yet) and Leon is edel-leon
- Then I checked the many Adam's and YES, the slugs differentiate with numbers since we don't have first names. So Adam, Adam-2, Adam-3. The only thing is that right now #11 is adam-2 and #12 is adam, even though they're not in that order. I think it's because #12 Adam only has a membership date in 1926, while #11 adam-2 has dates 1926-27. 

- YES I see the slug in the admin, as well as the text that explains what it is. See screenshot for Achenbach. 
![Screen Shot 2019-11-14 at 1 33 09 PM](https://user-images.githubusercontent.com/16673934/68885561-6597a000-06e3-11ea-8ab9-2c54ea6f949c.png)
- YES when I try to add a person with a slug that already exists, I get an error message and it tells me "Person with this Slug already exists" - I tried to enter in a new Achenbach and was chastised, as I should have been. 

- YES, I see the slug when I add in a new person. I added in "Rosef, Lily" and the slug was "rosef".
